File: app/src/main/java/net/afterday/compas/persistency/items/ItemDescriptor.java
package net.afterday.compas.persistency.items;

import net.afterday.compas.core.inventory.Inventory;
import net.afterday.compas.core.inventory.items.Item;
import net.afterday.compas.core.serialization.Jsonable;

/**
 * Created by Justas Spakauskas on 3/25/2018.
 */

public interface ItemDescriptor
{
    double NULL_MODIFIER = -99999999d;
    int getImage();
    String getName();
    int getNameId();
    boolean isBooster();
    boolean isDevice();
    boolean isArmor();
    boolean isArtefact();
    boolean isSingleUse();
    boolean isConsumable();
    long getDuration();
    double[] getModifiers();
    int getXpPoints();
    String getDescription();
    int getDescriptionId();
    boolean isUsable();
    boolean isDropable();
    Item.CATEGORY getCategory();
}

File: engine/src/main/kotlin/net/afterday/compas/engine/core/inventory/items/ItemDescriptor.kt
package net.afterday.compas.engine.core.inventory.items

import net.afterday.compas.engine.core.inventory.items.Item.CATEGORY

object ItemDescriptorConst {
    const val NULL_MODIFIER = -99999999.0
}

interface ItemDescriptor {
    val image: Int
    val name: String
    val nameId: Int
    val description: String
    val descriptionId: Int

    val isBooster: Boolean
    val isDevice: Boolean
    val isArmor: Boolean
    val isArtefact: Boolean

    val isSingleUse: Boolean
    val isUsable: Boolean
    val isDroppable: Boolean

    val isConsumable: Boolean
    val duration: Long

    val modifiers: DoubleArray
    val xpPoints: Int

    val category: CATEGORY?
}